### ISSUE TITLE
Efficient Neural Audio Synthesis

### ISSUE THREAD

**Issue by Swall0w** [Swall0w/papers#601]
Kalchbrenner, Nal, Elsen, Erich, Simonyan, Karen, Noury, Seb, Casagrande, Norman, Lockhart, Edward, Stimberg, Florian, Oord, Aaron van den, Dieleman, Sander, Kavukcuoglu, Koray

Sequential models achieve state-of-the-art results in audio, visual and
textual domains with respect to both estimating the data distribution and
generating high-quality samples. Efficient sampling for this class of models
has however remained an elusive problem. With a focus on text-to-speech
synthesis, we describe a set of general techniques for reducing sampling time
while maintaining high output quality. We first describe a single-layer
recurrent neural network, the WaveRNN, with a dual softmax layer that matches
the quality of the state-of-the-art WaveNet model. The compact form of the
network makes it possible to generate 24kHz 16-bit audio 4x faster than real
time on a GPU. Second, we apply a weight pruning technique to reduce the number
of weights in the WaveRNN. We find that, for a constant number of parameters,
large sparse networks perform better than small dense networks and this
relationship holds for sparsity levels beyond 96%. The small number of weights
in a Sparse WaveRNN makes it possible to sample high-fidelity audio on a mobile
CPU in real time. Finally, we propose a new generation scheme based on
subscaling that folds a long sequence into a batch of shorter sequences and
allows one to generate multiple samples at once. The Subscale WaveRNN produces
16 samples per step without loss of quality and offers an orthogonal method for
increasing sampling efficiency.

https://arxiv.org/abs/1802.08435
